crates/wasmparser/src/validator/operators.rs

```diff
@@ -196,10 +220,40 @@ impl OperatorValidator {
         Ok(())
     }
 
+    /// Returns the current operands stack height.
     pub fn operand_stack_height(&self) -> usize {
         self.operands.len()
     }
 
+    /// Returns the optional value type of the value operand at the given
+    /// `index` from the top of the operand stack.
+    ///
+    /// Returns `None` if the `index` is out of bounds.
+    ///
+    /// # Note
+    ///
+    /// An `index` of 0 will refer to the last operand on the stack.
+    pub fn peek_operand_at(&self, index: usize) -> Option<Option<ValType>> {
+        self.operands.iter().rev().nth(index).copied()
+    }
+
+    /// Returns the number of frames on the control flow stack.
+    pub fn control_stack_height(&self) -> usize {
+        self.control.len()
+    }
+
+    /// Returns a shared reference to the control flow [`Frame`] of the
+    /// control flow stack at the given `depth` if any.
+    ///
+    /// Returns `None` if the `depth` is out of bounds.
+    ///
+    /// # Note
+    ///
+    /// A `depth` of 0 will refer to the last frame on the stack.
```

Comment:
I think it's ok to skip the docs here on these functions since this `OperatorsValidator` isn't exposed publicly.

Reply:
In that these docs are all great to have but it's best I think to leave them on the main public methods for explanation and visibility

Reply:
Hmmm we are a bit flaky here since some of those internal methods do have quite some docs and others do not. I can remove the docs since you asked for that but now `control_stack_height` method has docs while `get_frame` has no docs ... :upside_down_face:

Reply:
Sorry I don't mean to say that we shouldn't have documentation, even for internals that don't show up in `rustdoc`. Moreso I prefer to avoid duplicate documentation where possible